camunda/v2/client.go

```diff
@@ -153,3 +154,57 @@ func (c *client) doPostRequest(ctx context.Context, params *bytes.Buffer, endpoi
 
 	return body, nil
 }
+
+func (c *client) DeleteTask(ctx context.Context, businessKey string) error {
+	tasks, err := c.getTasks(ctx, businessKey)
+	if err != nil {
+		return err
+	}
+	if len(tasks) == 1 {
+		if err = c.deleteProcessInstance(ctx, tasks[0].ProcessInstanceId); err != nil {
+			return err
+		}
+	} else {
+		return fmt.Errorf("found %d camunda tasks for businessKey: %s", len(tasks), businessKey)
+	}
+	return nil
+}
+
+func (c *client) getTasks(ctx context.Context, businessKey string) ([]Task, error) {
+	url := "task"
+	params := processTaskParams{
+		BusinessKey: businessKey,
+	}
+	buf := bytes.Buffer{}
+	err := json.NewEncoder(&buf).Encode(params)
+	if err != nil {
+		return nil, fmt.Errorf("failed to send camunda message due to json error: %w", err)
+	}
+
+	var tasks []Task
+	bytes, err := c.doPostRequest(ctx, &buf, url)
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(bytes, &tasks)
+	return tasks, err
```

Comment:
(not relevant): it's as a good practice to declare variables close to where they will be used
```suggestion
	bytes, err := c.doPostRequest(ctx, &buf, url)
	if err != nil {
		return nil, err
	}
	
	var tasks []Task
	err = json.Unmarshal(bytes, &tasks)
	return tasks, err
```